activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -95,7 +97,7 @@ def assert_difference(expression, *args, &block)
         }
         before = exps.map(&:call)
 
-        retval = yield
+        retval = assert_nothing_raised { yield }
```

Comment:
```suggestion
        retval = assert_nothing_raised(&block)
```

This is quicker these days and we already capture the block in the method signature.